lib/timers.js

```diff
@@ -554,67 +557,73 @@ function tryOnImmediate(immediate, queue) {
   }
 }
 
+function runCallback(timer) {
+  var argv = timer._argv;
+  var argc = argv ? argv.length : 0;
+  switch (argc) {
+    // fast-path callbacks with 0-3 arguments
+    case 0:
+      return timer._callback();
+    case 1:
+      return timer._callback(argv[0]);
+    case 2:
+      return timer._callback(argv[0], argv[1]);
+    case 3:
+      return timer._callback(argv[0], argv[1], argv[2]);
+    // more then 3 arguments run slower with .apply
+    default:
+      return timer._callback.apply(timer, argv);
+  }
+}
 
-function Immediate() { }
-
-Immediate.prototype.domain = undefined;
-Immediate.prototype._onImmediate = undefined;
-Immediate.prototype._idleNext = undefined;
-Immediate.prototype._idlePrev = undefined;
 
+function createImmediate(callback, args) {
```

Comment:
How did you test?  I'm still seeing a very consisten difference with the sources from
this pull request.  The following test can be run standalone to show the difference:
```
var n = 0;
function counter() { n += 1; }

var count = 4000000;
var t1 = Date.now();
for (var i=0; i<count; i++) setImmediate(counter);
setImmediate(function() {
    var t2 = Date.now();
    console.log("ran %dk setImmediate in %d ms (n calls)", count/1000, (t2 - t1), n + 1, process.memoryUsage());
})
```
createImmediate version: (run with `$ taskset 8 ./node time-setImmediate.js`):
ran 1000k setImmediate in 301 ms (n calls) 1000001 { rss: 64249856, heapTotal: 51257344, heapUsed: 38976796 }
new Immediate version:
ran 1000k setImmediate in 5295 ms (n calls) 1000001 { rss: 97222656, heapTotal: 82714624, heapUsed: 66396932 }

This is much more extreme a difference than in the other tests, but it's consistent.
The rss and heap footprint is much larger, I'm seeing a lot of calls to futex() and 12k
context switches / second while it's running.  I don't know what it's doing to run so slow.
On other tests I'm seeing a steady 15% difference.

The "other test" is the one-liner from a larger suite
```
case '4a1':
  runit(nloops, 200, 10000, "set and run 10k setImmediate, 1 arg", function(cb) { for (var i=0; i<10000; i++) setImmediate(f, 1); setImmediate(cb) });
  break;
```
I packaged up my timeit / runit utility as `qtimeit`, so the test can be run in isolation
```
timeit = require('qtimeit');
function f(){}
timeit.runit(10, 200, 10000, "set and run 10k setImmediate, 1 arg", function(cb) { for (var i=0; i<10000; i++) setImmediate(f, 1); setImmediate(cb) });
```
createImmediate: `200 loops in 0.5001 sec: 399.92 / sec,`
new Immediate: `200 loops in 0.5859 sec: 341.35 / sec,`